src/common/meta/src/frontend_expiry.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Mutex;
+use std::time::Duration;
+
+use common_telemetry::{debug, error, info, warn};
+use tokio::task::JoinHandle;
+use tokio::time::{interval, MissedTickBehavior};
+
+use crate::cluster::{NodeInfo, NodeInfoKey, Role};
+use crate::error;
+use crate::kv_backend::ResettableKvBackendRef;
+use crate::leadership_notifier::LeadershipChangeListener;
+use crate::rpc::store::RangeRequest;
+use crate::rpc::KeyValue;
+
+/// FrontendExpiryListener periodically checks all node info in memory and removes
+/// expired node info to prevent memory leak.
+pub struct FrontendExpiryListener {
+    handle: Mutex<Option<JoinHandle<()>>>,
+    tick_interval: Duration,
+    max_idle_time: Duration,
+    in_memory: ResettableKvBackendRef,
+}
+
+impl Drop for FrontendExpiryListener {
+    fn drop(&mut self) {
+        self.stop();
+    }
+}
+
+impl FrontendExpiryListener {
+    pub fn new(
+        tick_interval: Duration,
+        max_idle_time: Duration,
+        in_memory: ResettableKvBackendRef,
+    ) -> Self {
+        Self {
+            handle: Mutex::new(None),
+            tick_interval,
+            max_idle_time,
+            in_memory,
+        }
+    }
+
+    async fn start(&self) {
+        let mut handle = self.handle.lock().unwrap();
+        if handle.is_none() {
+            let tick_interval = self.tick_interval;
+            let in_memory = self.in_memory.clone();
+
+            let max_idle_time = self.max_idle_time;
+            let ticker_loop = tokio::spawn(async move {
+                let mut interval = interval(tick_interval);
+                interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+                loop {
+                    interval.tick().await;
+                    if let Err(e) =
+                        Self::clean_expired_frontend_node(&in_memory, max_idle_time).await
+                    {
+                        error!(e; "Failed to clean expired frontend node");
+                    }
+                }
+            });
+            *handle = Some(ticker_loop);
+        }
+    }
+
+    fn stop(&self) {
+        if let Some(handle) = self.handle.lock().unwrap().take() {
+            handle.abort();
+            info!("Frontend expiry ticker stopped")
+        }
+    }
+
+    /// Cleans expired nodes from memory.
+    async fn clean_expired_frontend_node(
```

Comment:
```suggestion
    async fn clean_expired_frontend_nodes(
```

Reply:
fixed in https://github.com/GreptimeTeam/greptimedb/pull/5592/commits/ccb40c67c5b4b1d089fc0a6ac25ff583b957f42e#diff-01d75c35b8bc2fba5b556a3b1aa86e7fa2bfb713eebe2ac5db5b3116fb1d55c5R81